src/app/clusters/on-off-server/on-off-server.cpp

```diff
@@ -613,15 +613,14 @@ bool OnOffServer::areStartUpOnOffServerAttributesNonVolatile(EndpointId endpoint
  */
 EmberEventControl * OnOffServer::getEventControl(EndpointId endpoint)
 {
-    uint16_t index            = emberAfFindClusterServerEndpointIndex(endpoint, OnOff::Id);
-    EmberEventControl * event = nullptr;
+    static EmberEventControl eventControls[EMBER_AF_ON_OFF_CLUSTER_SERVER_ENDPOINT_COUNT];
```

Comment:
This incurs extra codesize to check for init on every call, no?  Please leave `eventControls` where it was.  If you think it's worth moving, that should be a separate PR.

Is the problem that the array sizing depends on endpoint_config and that's a problem for the header include?  If so, making this a file static is probably better than having a function static....

Reply:
The problem is indeed the header declaration depending on the endpoint count. I had started with a file static but saw that it is only used in a single function so I switched it to function static.

I am quite sure the storage requirement for all 3 cases is identical (except maybe the function static would be optimized out if the function is never called). This should not incur any extra storage on any of the function calls.

Reply:
It's not about storage, it's about checking whether the static has already been initialized and initializing it if not....

Reply:
I would have expected some init code too, however generally since these are BSS zero-sized it probably wont happen:

![image](https://user-images.githubusercontent.com/1832280/220400871-54550cfd-8321-4a98-9cd2-9af39985b683.png)

However to not worry about it, switched this to a file static now.

Reply:
Current UI lets the app instantiate it, so app can instantiate it on the endpoints it needs. It is indeed not based on codegen: one would need to change main to match generated code which is some duplication, however more generic.

It is assumed that the app knows exactly how many endpoints (and which ones) have light on it.